frontend/src/components/GamemodeBar.jsx

```diff
@@ -20,95 +20,152 @@ function GamemodeBar() {
+                <GamemodeButton
+                  className={gamemode.type === "Impossible" ? "active" : ""}
                   onClick={() => changeGamemode("Circlefall", "Impossible")}
                 >
                   Impossible
-                </button>
+                </GamemodeButton>
 
-                <button
-                  className={`gamemode-button ${gamemode.type === "Marathon" ? "active" : ""}`}
+                <GamemodeButton
+                  className={gamemode.type === "Marathon" ? "active" : ""}
                   onClick={() => changeGamemode("Circlefall", "Marathon")}
                 >
                   Marathon
-                </button>
+                </GamemodeButton>
 
-              </div>
+              </GamemodeDataButtons>
             </>
           )}
 
           {gamemode.mode === "Gridshot" && (
             <>
-              <div className="divider"></div>
-
-              <div className="gamemode-data-buttons">
+              <GamemodeDataButtons>
 
-              <button
-                  className={`gamemode-button ${gamemode.type === "Classic" ? "active" : ""}`}
+              <GamemodeButton
+                  className={gamemode.type === "Classic" ? "active" : ""}
                   onClick={() => changeGamemode("Gridshot", "Classic")}
                 >
                   Classic
-                </button>
-                <button
-                  className={`gamemode-button ${gamemode.type === "Mini" ? "active" : ""}`}
+                </GamemodeButton>
+                <GamemodeButton
+                  className={gamemode.type === "Mini" ? "active" : ""}
                   onClick={() => changeGamemode("Gridshot", "Mini")}
                 >
                   Mini
-                </button>
+                </GamemodeButton>
 
-              </div>
+              </GamemodeDataButtons>
             </>
           )}
 
-        </div>
-      </div>
+        </GamemodeBarDiv>
+      </GamemodeBarContainer>
 
       </>
     )
 }
 
-export default GamemodeBar
\ No newline at end of file
+export default GamemodeBar;
+
+//Styling
+const GamemodeBarContainer = styled.div`
+  grid-row: 2 / 3;
+  display: flex;
+  justify-content: center;
+  margin-top: 5px;
+`;
+
+const GamemodeBarDiv = styled.div`
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  padding: 5px;
+  background-color: #2b2b2b;
+  border-radius: 10px;
+  transition: opacity 0.2s ease-in-out, visibility 0.2s ease-in-out;
+
+  &.hidden {
+    opacity: 0;
+    visibility: hidden;
+  }
+`;
+
+const GamemodeDataButtons = styled.div`
+  display: inline-flex;
+`
+
+const GamemodeButton = styled.button`
+  background-color: transparent;
+  color: #666666;
+  border: none;
+  padding: 8px 20px;
+  margin: 0 5px;
+  border-radius: 5px;
+  outline: none;
+  transition: color 0.3s, transform 0.3s;
+
+  &:hover{
```

Comment:
Nice job figuring this out! Definitely thankful that the class stuff was still possible